core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1882,12 +1883,8 @@ protected void _run() throws IOException, InstallationStatus {
          */
         protected void replace(File dst, File src) throws IOException {
             File bak = Util.changeExtension(dst,".bak");
-            bak.delete();
-            dst.renameTo(bak);
-            dst.delete(); // any failure up to here is no big deal
```

Comment:
Note: behavioral change.

Reply:
Yeah, this seemed like more straightforward behavior to me.